pcdsdevices/pneumatic.py

```diff
@@ -0,0 +1,98 @@
+"""
+Pneumatic Classes.
+
+This Module contains all the classes relating to Pneumatic Actuators
+"""
+
+from lightpath import LightpathState
+from ophyd import Component as Cpt
+from ophyd.status import Status
+
+from pcdsdevices.interface import BaseInterface, LightpathMixin
+
+from .signal import PytmcSignal
+
+
+class BeckhoffPneumatic(BaseInterface, LightpathMixin):
+    """
+    Class containing basic Beckhoff Pneumatic support
+    """
+    lightpath_cpts = ['limit_switch_in', 'limit_switch_out']
+
+    # readouts
+    limit_switch_in = Cpt(PytmcSignal, ':PLC:bInLimitSwitch', io="i")
+    limit_switch_out = Cpt(PytmcSignal, ':PLC:bOutLimitSwitch', io="i")
+
+    retract_status = Cpt(PytmcSignal, ':bRetractDigitalOutput', io="i")
+    insert_status = Cpt(PytmcSignal, ':bInsertDigitalOutput', io="i")
+
+    # logic and supervisory
+    interlock_ok = Cpt(PytmcSignal, ':bInterlockOK', io="i")
+    insert_ok = Cpt(PytmcSignal, ':bInsertEnable', io="i")
+    retract_ok = Cpt(PytmcSignal, ':bRetractEnable', io="i")
+
+    # commands
+    insert_signal = Cpt(PytmcSignal, ':CMD:IN', io="io")
+    retract_signal = Cpt(PytmcSignal, ':CMD:OUT', io="io")
+
+    # returns
+    busy = Cpt(PytmcSignal, ':bBusy', io="i")
+    done = Cpt(PytmcSignal, ':bDone', io="i")
+    reset = Cpt(PytmcSignal, ':bReset', io="io")
+    error = Cpt(PytmcSignal, ':PLC:bError', io="i")
+    error_id = Cpt(PytmcSignal, ':PLC:nErrorId', io="i")
+    error_message = Cpt(PytmcSignal, ':PLC:sErrorMessage', io="i")
```

Comment:
It looks to me like this is returning as an array of ints instead of as a string, probably because the PV is a char waveform
```suggestion
    error_message = Cpt(PytmcSignal, ':PLC:sErrorMessage', io="i", string=True)
```